buidl/script.py

```diff
@@ -101,10 +104,18 @@ def parse(cls, stream):
                 commands.append(op_code)
         obj = cls(commands)
         if count != length:
+            # FIXME: shouldn't this throw an error?
             print(f"mismatch between length and consumed bytes {count} vs {length}")
             obj.raw = raw
         return obj
 
+    @classmethod
+    def parse_hex(cls, hex_str, is_p2sh_multisig=False):
```

Comment:
This is why stream.read() will work up top. If this is a normal stream, it won't work.